client/boot/common.js

```diff
@@ -237,10 +238,28 @@ const configureReduxStore = ( currentUser, reduxStore ) => {
 };
 
 function setupErrorLogger( reduxStore ) {
+	// Add a bit of metadata from the redux store to the sentry event.
+	const beforeSend = ( event ) => {
+		const state = reduxStore.getState();
+		if ( ! event.tags ) {
+			event.tags = {};
+		}
+		event.tags.blog_id = getSelectedSiteId( state );
+		event.tags.calypso_section = getSectionName( state );
+		return event;
+	};
+
+	// Note that Sentry can disable itself and do some cleanup if needed, so we
+	// run it before the catch-js-errors check. (Otherwise, cleanup would never
+	// never happen.)
+	initSentry( { beforeSend } );
```

Comment:
Should we move this call after the condition that checks if the `catch-js-errors` config is `true`? It doesn't seem to make sense to init Sentry here and `return` just afterwards, while there's still code that should run that Sentry depends on (the one that's part of the `page('*')` route, so I assume we should *not* enable Sentry if `catch-js-errors` is `false`.

Reply:
I tried to explain in the comment, but it could definitely be clearer! More specifically, the way the async-loaded sentry package works, anything in Calypso should be able to import and call our stubbed Sentry functions. This way, we can utilize sentry functions like "add breadcrumb" anywhere in the Calypso application without wrapping it in a conditional -- because as we know, Sentry might not always be loaded.

Internally, those functions will queue up calls to sentry as long as sentry is not loaded. As a result, if we know we're _never_ going to load sentry, that queue could (theoretically) get pretty big because the queue is never processed. So if we know we aren't going to use sentry at all, those stubbed functions should just return and not do anything -- and the queue of calls should be cleared once we know sentry won't be loaded.

When `initSentry` is called, it checks all of the factors -- such as the RNG, `catch-js-errors`, etc. If it determines we shouldn't load sentry, it clears the queues and exits. It also sets its status to "disabled", which causes the stubbed functions to just return without doing anything. So at that point, the stubbed functions can be called anywhere in Calypso, but Sentry isn't loaded and they don't do anything, minimizing the bundle size impact.

In other words, it's designed to called _always_, and it conditionally disables itself internally.

I could also rewrite this to be more explicit if the approach isn't very clear 🤔

Reply:
Ah, I see, thanks for the comprehensive explanation! 👍🏻